File: solutions/26. Remove Duplicates from Sorted Array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        // auto itr = unique(nums.begin(), nums.end());
        // nums.resize(distance(nums.begin(), itr));
        // return nums.size();
        
        int n = nums.size();
        if (n < 2) return n;
        
        int tmp = nums[0];
        int ret = 1;
        for (int i = 1; i < n; i++) {
            if (tmp != nums[i]) {
                nums[ret] = nums[i];
                ret++;
                
                tmp = nums[i];
            }
        }
        
        return ret;
    }
};

File: ..solutions/26. Remove Duplicates from Sorted Array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        return optimized(nums);
    }
​
    private int optimized(int[] nums) {
        int n=nums.length;
        int left=-1;
        int right=0;
        while (right<n){
            int elem=nums[right];
            if (left==-1||nums[left] != nums[right])
                nums[++left]=nums[right];
            right++;
        }
        return left+1;
    }
}
